### ISSUE TITLE
Unable to get new address rpcclient.GetNewAddress() for simnet due to hardcoded mainnet params

### ISSUE THREAD

**Issue by nostdm** [btcsuite/btcd#1519]
Some wallet methods in the `rpcclient` library have hardcoded the `chaincfg.MainNetParams` for example the `GetNewAddress()` method calls the `Receive()` method which hardcodes the `MainNetParams` when decoding the address:

```
func (r FutureGetNewAddressResult) Receive() (btcutil.Address, error) {
	res, err := receiveFuture(r)
	if err != nil {
		return nil, err
	}

	// Unmarshal result as a string.
	var addr string
	err = json.Unmarshal(res, &addr)
	if err != nil {
		return nil, err
	}

	return btcutil.DecodeAddress(addr, &chaincfg.MainNetParams)
}
```

This returns an error when talking to btcwallet/btcd in `--simnet` mode.

Is there a way to tell the `rpcclient` to use `SimnetParams` when talking to a simnet server?

Potential way to fix this would be to allow the user to specify the network type when constructing the client:
```
	client, err := rpcclient.New(&rpcclient.ConnConfig{
		Host: ...,
		Endpoint: ...,
		User: ...,
		Pass: ...,
		Certificates: ...,
                Network: "Simnet",
	}, &notificationHandlers)
```

**Comment by torkelrogstad** [comment]
Duplicate of https://github.com/btcsuite/btcd/issues/1466

**Comment by nostdm** [comment]
It seems that this PR fixes the issue https://github.com/btcsuite/btcd/pull/1467 Is there any reason it hasn't been merged since November?

**Comment by jakesyl** [comment]
Fixed by #1467
